### PR TITLE
Add common_ancestor function

### PR DESCRIPTION
**Pull request by mkborregaard**
Adds a new metrics.jl file which currently only has a function to identify the most recent common ancestor. See discusion at #40. 

@tpoisot I've branched off your master branch, is that OK? 
@isaacovercast would you mind testing?

### CODE REVIEW

**Comment by isaacovercast** [comment]
@mkborregaard Tested and it works, or at least it agrees with the MRCA for the code I wrote. Thanks!

**Comment by mkborregaard** [comment]
How do the speeds stack up in a real-world example?

**Comment by isaacovercast** [comment]
Yours is faster ;)

Mine:
```
@benchmark getmrca(tre, names, false)
BenchmarkTools.Trial: 
  memory estimate:  25.05 KiB
  allocs estimate:  321
  --------------
  minimum time:     40.762 μs (0.00% GC)
  median time:      45.852 μs (0.00% GC)
  mean time:        49.266 μs (5.21% GC)
  maximum time:     8.681 ms (99.20% GC)
  --------------
  samples:          10000
  evals/sample:     1
```
Yours:
```
@benchmark common_ancestor(tre, names)
BenchmarkTools.Trial: 
  memory estimate:  3.58 KiB
  allocs estimate:  31
  --------------
  minimum time:     6.989 μs (0.00% GC)
  median time:      7.741 μs (0.00% GC)
  mean time:        8.462 μs (4.22% GC)
  maximum time:     1.822 ms (99.05% GC)
  --------------
  samples:          10000
  evals/sample:     4
```

**Review by richardreeve — CHANGES_REQUESTED** [review]
I think the test may fail if we change anything about the parsing, so I've suggested a change. The other issue is with naming - I've no problem in principle with `common_ancestor`, but it's not consistent with what we've done in the rest of the package - I'd go for `commonancestor` for `mrca` probably...

File: src/Phylo.jl [hunk 1]
```diff
@@ -181,6 +181,10 @@ export droptips!, keeptips!
 # Plot recipes
 include("plot.jl")
 
+# Metrics from the tree
+include("metrics.jl")
+export common_ancestor
```
[the remark this answers is not in the document]
Comment (mkborregaard) [reply src/Phylo.jl:186]: `commonancestor` is hard to read though. Maybe `mrca` is well-established enough as a term?

[the remark this answers is not in the document]
Comment (mkborregaard) [reply src/Phylo.jl:186]: I'll change it to whatever you suggest @richardreeve and then go ahead and merge if that's OK

[the remark this answers is not in the document]
Comment (mkborregaard) [reply src/Phylo.jl:186]: This seems to indicate that `mrca`should work
Skærmbillede 2021-01-14 kl  00 11 03

[the remark this answers is not in the document]
Comment (richardreeve) [reply src/Phylo.jl]: I think mrca is fine for anyone doing phylogenetics. Tbh, when I saw "common ancestor", I had to translate it into mrca in my head!

**Review by richardreeve — APPROVED** [review]

**Comment by mkborregaard** [comment]
tests fail on 1.0 because of JuliaDB - do we care?

**Comment by richardreeve** [comment]
No, @claireh93 is preparing a separate PR part of which will remove the 1.0 tests.

**Pull request merged**